resolwe_bio/processes/WGBS/walt_index.py

```diff
@@ -0,0 +1,73 @@
+"""Create genome index for WALT aligner."""
+import shutil
+
+from pathlib import Path
+from plumbum import TEE
+from resolwe.process import Cmd, DataField, FileField, Process, StringField, DirField
+
+
+class WaltIndex(Process):
+    """Create WALT genome index."""
+
+    slug = 'walt-index'
+    process_type = 'data:index:walt'
+    name = 'WALT genome index'
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/rnaseq:4.9.0'
+            },
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 16384,
+        },
+    }
+    category = 'Genome index'
+    data_name = '{{ ref_seq.file|default("?") }}'
+    version = '1.0.0'
+
+    class Input:
+        """Input fields for WaltIndex."""
+
+        ref_seq = DataField('seq:nucleotide', label='Reference sequence (nucleotide FASTA)')
+
+    class Output:
+        """Output fields to process WaltIndex."""
+
+        index = DirField(label='WALT index')
+        fastagz = FileField(label='FASTA file (compressed)')
```

Comment:
Wy do we need fasta and fastagz outputs here? Isn't is ufficient to have them in the FASTA Data object? This duplicates the files and also disk usage...

Reply:
I agree that this is not efficient in terms of storage. However:
- none of the currently supported aligners need .fasta for the alignment job. However, this might not always be the case. It's easier / less costly to have the input fasta next to the index now, than to handle this in some future migration.
- A typical sample will now have an index object on the input. Consider a scenario, where you want to see the resulting .bam in a genome browser. For this you also need the reference sequence. If the .fasta is not attached to the index object, one would need to fetch a reference sequence through additional parent/child relations.
- Existing workflows often take a current ``data:genome:fasta`` on the input. Often, both the index and the fasta are used in (multiple) workflow steps. By keeping the .fasta next to the index, we can retain only one "genome" input in most of the workflows.
- The number of actual index objects is relatively small - the storage problem is not that great in practice.

Reply:
OK, thanks for the explanation! Agree with you on all points, this makes sense.